staging/src/github.com/kubeedge/beehive/pkg/core/core.go

```diff
@@ -84,3 +90,46 @@ func moduleKeeper(name string, moduleInfo *ModuleInfo, m common.ModuleInfo) {
 		beehiveContext.AddModuleGroup(name, moduleInfo.module.Group())
 	}
 }
+
+// localModuleKeeper starts and tries to keep module running when module exited.
+// Call EnableModuleRestart() to enable auto-restarting feature in alpha version.
+func localModuleKeeper(m *ModuleInfo) {
+	if !moduleRestartEnabled {
+		m.module.Start()
+		return
+	}
+
+	ctx := beehiveContext.GetContext()
+	backoffDuration := time.Second
+
+	// do if module exits
+	afterFunc := func() {
+		if r := recover(); r != nil {
+			klog.Errorf("module %s panicking: %v", m.module.Name(), r)
+		}
+		m.active = false
```

Comment:
Where else is this field `active` used? Seems only here and L117

Reply:
It hasn't been used now. I'll remove it.